### PR TITLE
[PW-529]: Dynamic origin key support

### PR DESCRIPTION
**Pull request by rkewlani**
**Description**

**Tested scenarios**

**Fixed issue**:

### CODE REVIEW

File: adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java [hunk 1]
```diff
@@ -223,10 +223,15 @@ public void validateHPPResponse(final HttpServletRequest request) throws Signatu
...
+    public String getOriginKey(HttpServletRequest request) throws IOException, ApiException {
+        return getAdyenPaymentService().getOriginKey(getBaseURL(request));
+    }
 
-        return baseStore.getAdyenOriginKey();
+    public String getBaseURL(HttpServletRequest request) {
+        String currentRequestURL = request.getRequestURL().toString();
+        int requestUrlLength = currentRequestURL.length();
+        int requestUriLength = request.getRequestURI().length();
+        return currentRequestURL.substring(0, requestUrlLength - requestUriLength);
```
Comment (rikterbeek) [inline adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java]: do we have port support ?

File: adyenv6core/src/com/adyen/v6/service/DefaultAdyenPaymentService.java [hunk 2]
```diff
@@ -328,6 +332,23 @@ public PaymentsResponse getPaymentDetailsFromPayload(HashMap<String, String> det
...
 
+    @Override
+    public String getOriginKey(String originDomain) throws ApiException, IOException {
+        CheckoutUtility checkoutUtility = new CheckoutUtility(client);
+        OriginKeysRequest originKeysRequest = new OriginKeysRequest();
+        String originkey = "";
+        ArrayList<String> originDomains = new ArrayList<>(Arrays.asList(originDomain));
+
+        originKeysRequest.setOriginDomains(originDomains);
+        LOG.debug(originKeysRequest);
```
Comment (rikterbeek) [inline adyenv6core/src/com/adyen/v6/service/DefaultAdyenPaymentService.java]: is this not already logged in the library ?

File: adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java (hunk 1 above)
Comment (rkewlani) [reply adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java]: yes

File: adyenv6core/src/com/adyen/v6/service/DefaultAdyenPaymentService.java (hunk 2 above)
Comment (rkewlani) [reply adyenv6core/src/com/adyen/v6/service/DefaultAdyenPaymentService.java]: no, we don't do any logging in library.

**Pull request merged**